draft/jumpstarter.rst

```diff
@@ -0,0 +1,218 @@
+.. vale off
+
+======================
+CEP XXXX: Jumpstarter
+======================
+
+:CEP: XXXX
+:Author: Micah Lyle
+:Implementation Team: Omer Katz
+:Shepherd: Omer Katz
+:Status: Draft
+:Type: Feature
+:Created: 2021-05-09
+:Last-Modified: 2021-05-09
+
+.. contents:: Table of Contents
+   :depth: 3
+   :local:
+
+Abstract
+========
+
+`Next-Gen Celery`_ aims to:
+
+1. Move Celery from the past to the present. 
+
+2. Resolve long-standing design bugs in our implementation. 
+
+3. Modernize the code to support Python3+. 
+
+4. Do a number of other things not entirely in the scope of this document (see `Next Gen Rationale`_). 
+
+Before work is to begin on this next generation of Celery, it's crucial to have
+a discussion about and carve out the building blocks that will form the
+primitives for the upcoming Celery. These building blocks should fulfill 1, 2,
+and 3 (especially) above, and also help fulfill the goals from the linked
+`Next-Gen Rationale`_.
+
+For Celery to move forward into the modern age, and the future of computing
+with increasingly decentralized systems, it's important for Celery to follow a
+standardized asynchronous execution model. Namely, we want to define things
+like: 
+
+1. How does Celery interact with itself (its own internal components)? 
+
+2. How does Celery interact with others (and how do others interact with it)? 
+
+3. How do Celery's internal components interact with each other and their outside environment? 
+
+Also, we want Celery to work with the emerging and maturing asynchronous python
+landscape.  Celery is *very asynchronous* by nature, dealing with brokers,
+queues, results, timeouts, chord-like joins, task groups, you name it.
+Modern asynchronous python frameworks have provided new, innovative, and robust
+solutions to many things that overlap with Celery's goals.
+
+To do all of this, we propose modeling next-gen Celery off of the `Actor Model`_. Namely,
+we propose that the Celery Worker will be modeled as an `Actor System`_ of sorts.
+
+By the end of this initial implementation, we should have a proposal and reference implementation that: 
+
+1. Models Tasks and other key Celery primitives (some to be implemented and
```

Comment:
I think we should reduce scope to first get a working Actor framework with all the features it needs to function as a standalone framework and then starts modeling Celery primitives as actors.

Reply:
Agreed. Working on that next. The next draft should have that.

Reply:
I think the trickiest part at the moment for me is trying to figure out how much to talk about Jumpstarter and how much to talk about Celery. Do you feel like this CEP should be more focused on the whys and what's of a working Actor framework in Python, or the whys and whats of a future async Celery and the rationale for building an underlying standalone framework that would "power" Celery (but also contribute something to the broader Python community in terms of a missing actor framework).?